src/main/java/edu/cornell/kfs/tax/service/impl/TaxParameterServiceImpl.java

```diff
@@ -0,0 +1,165 @@
+package edu.cornell.kfs.tax.service.impl;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.Validate;
+import org.kuali.kfs.coreservice.framework.parameter.ParameterService;
+import org.kuali.kfs.coreservice.impl.parameter.Parameter;
+import org.kuali.kfs.sys.KFSConstants;
+import org.springframework.cache.annotation.Cacheable;
+
+import edu.cornell.kfs.sys.CUKFSConstants;
+import edu.cornell.kfs.tax.CUTaxConstants;
+import edu.cornell.kfs.tax.service.TaxParameterService;
+
+public class TaxParameterServiceImpl implements TaxParameterService {
+
+    private ParameterService parameterService;
+
+    @Override
+    public Parameter getParameter(final String componentCode, final String parameterName) {
+        validateParams(componentCode, parameterName);
+        return parameterService.getParameter(CUTaxConstants.TAX_NAMESPACE, componentCode, parameterName);
+    }
+
+    @Override
+    public String getParameterValueAsString(final String componentCode, final String parameterName) {
+        validateParams(componentCode, parameterName);
+        return parameterService.getParameterValueAsString(CUTaxConstants.TAX_NAMESPACE, componentCode, parameterName);
+    }
+
+    @Override
+    public Collection<String> getParameterValuesAsString(final String componentCode, final String parameterName) {
+        validateParams(componentCode, parameterName);
+        return parameterService.getParameterValuesAsString(CUTaxConstants.TAX_NAMESPACE, componentCode, parameterName);
+    }
+
+    @Override
+    @Cacheable(cacheNames = Parameter.CACHE_NAME, key = "'{CU_getParameterValuesSetAsString}'+#p0+','+#p1")
+    public Set<String> getParameterValuesSetAsString(final String componentCode, final String parameterName) {
```

Comment:
Is the Cacheable required or necessary in this clas? Does it provide needed performance benefits?

Reply:
There are various inclusion/exclusion checks that need to be performed on all or most of the relevant Transaction Detail rows, where many of those checks involve comparisons against parameter values. Because the checks are much easier to perform when the parameters have been parsed into Sets/Maps/etc., and because it's inefficient to recreate those collections for each data row iteration, I wanted to add a way to temporarily cache the generated Sets/Maps/etc. without necessarily storing them in a temporary helper object. Putting the temporary storage at the Spring Cache level (backed by Redis in our case) seemed like the best way to accomplish that.